### DISCUSSION TITLE
Method group for parameter-less methods.

### DISCUSSION THREAD

**Discussion by MkazemAkhgary** [dotnet/csharplang#921] [General]
This is 😕 to me, but lets give it a shot. allow method groups for parameter less methods. for example `x => x.Foo()` becomes `.Foo`. 

Unfortunately `.` cant be omitted because it can cause ambiguity if field with same name already exist so that would be a breaking change. 

```
    listOfLists.Where(.Any);
```

Would be kinda cool but you see these weird syntaxes more likely in c++ than in c# 😄.

**Comment by jnm2** [comment]
See https://github.com/dotnet/csharplang/issues/91

**Comment by MkazemAkhgary** [comment]
@jnm2 cool, didn't saw that. hmm I don't see necessity of @ by the way.

**Comment by jnm2** [comment]
Check out the other issues linked to #91 too.

**Comment by bondsbw** [comment]
@MkazemAkhgary `@` would be used with `.Any()` instead of `.Any`.  So instead of being like a method group, it is a placeholder which can be used multiple times (e.g.  `listOfInts.Where(@ >= 0 && @ < 100)`

**Comment by quinmars** [comment]
@MkazemAkhgary, `Any()` is technically not a parameterless method, but an extension method. So the following already works:

```csharp
var res = listOfLists.Where(Enumerable.Any);
```

but that's not really shorter than:

```csharp
var res = listOfLists.Where(l => l.Any());
```

**Comment by RogerWaters** [comment]
@quinmars you can use static imports to import 'Enumerable.Any' than code is short:
```csharp
var res = listOfLists.Where(Any);
```

**Comment by quinmars** [comment]
@RogerWaters nice idea! This would even be shorter. It does not work, though, for `Any()` because `Any()` is an extension method.

IMHO, lambda expressions are compact enough and I do not need any obscure shortening here.
